llvm/lib/Target/AArch64/GISel/AArch64RegisterBankInfo.cpp

```diff
@@ -407,6 +411,14 @@ void AArch64RegisterBankInfo::applyMappingImpl(
             OpdMapper.getInstrMapping().getID() <= 4) &&
            "Don't know how to handle that ID");
     return applyDefaultMapping(OpdMapper);
+  case TargetOpcode::G_INSERT_VECTOR_ELT: {
+    // Extend smaller gpr operands to 32 bit.
+    Builder.setInsertPt(*MI.getParent(), MI.getIterator());
+    auto Ext = Builder.buildAnyExt(LLT::scalar(32), MI.getOperand(2).getReg());
```

Comment:
Could you move this code into the legalizer? We only state legal types for index 0. There are no references to 1 and 2.

Reply:
This can't be part of the legalizer for the reasons already mentioned already - the types we want depend on which register bank was selected for the inserted element.

Reply:
MIR after the legalizer must be legal. There cannot be i8 or i16 types. They are plain illegal. We have to fix the legalizer to prevent illegal types.